Android UI & API Improvements

jstheoriginal opened the pull request:
Fixed up the conflicts from https://github.com/expo/react-native-action-sheet/pull/29.

However, I didn't realize I couldn't use this without expo...which my app doesn't use. :( So if someone else wants to confirm this is good, feel free to merge it. :)

brentvatne commented:
why can't you use this without expo? there is no expo-specific code in https://github.com/expo/react-native-action-sheet/blob/master/ActionSheet.android.js

brentvatne commented:
cc @peterpme

flex-tjon commented:
I tested this out in our own app; this is what the title looks like on android:
![expo action sheet android example](https://user-images.githubusercontent.com/27235334/37800347-bca49288-2ddf-11e8-84ed-9ce457a904e3.png)

peterpme commented:
I will give it a go in just a bit!

peterpme commented:
@flex-tjon whats it supposed to look like?

peterpme commented:
@flex-tjon is the title supposed to be gray?

peterpme commented:
@jstheoriginal great work!

- Do you mind making the title & message gray?
- Do you also mind getting rid of the separators? I don't believe they're supposed to be there (unless somebody knows better)

The reference image is located here: https://material.io/guidelines/components/bottom-sheets.html#bottom-sheets-persistent-bottom-sheets

![image](https://user-images.githubusercontent.com/1211905/37801200-de026060-2de2-11e8-8d80-c61ce8a53707.png)

Thanks!

flex-tjon commented:
@peterpme `message` is an `ActionSheetIOS` [prop](https://facebook.github.io/react-native/docs/actionsheetios.html), so this keeps the android/iOS implementations in sync

peterpme commented:
Fair enough!

maxaggedon commented:
Any plans on merging this one ?

jstheoriginal commented:
@brentvatne I can’t recall why I didn’t think this would work in my non-expo app. Maybe I just assumed it wouldn’t since it was part of the expo repo (or maybe I tried to use the decorator and it wouldn’t work, not realizing I could do the other export default way).

I could work on this if still desired.

I could make the grey changes but then also add a textStyle prop so people could make overrides.

Same goes for the dividers. They could be a prop like `showDividers` and default to false.

maxaggedon commented:
@jstheoriginal it would be great! Let me know if I can help.

@peterpme is this actionsheet supposed to strictly follow material guidelines?

peterpme commented:
@maxaggedon we'd like to get as close as possible to native functionality for both ios / android. 

@jstheoriginal I'm sure there are *a ton* of folks interested in so I would encourage you to make it happen! 

It would be awesome if we could close this out :)

jstheoriginal commented:
Alright...I've given it my best go. Give it a look through. I've changed a lot...so let me know if you want changes.

Some screenshots showing a bunch of different options (which I added to the example app):

![screenshot_1532672336](https://user-images.githubusercontent.com/5117473/43305342-0a74de3a-913d-11e8-80d1-cdab3a52a3e2.png)

![screenshot_1532672323](https://user-images.githubusercontent.com/5117473/43305344-0eb05768-913d-11e8-9e1c-e3f53019350b.png)

![screenshot_1532672317](https://user-images.githubusercontent.com/5117473/43305347-119955ba-913d-11e8-88aa-924d347184a5.png)

![screenshot_1532672331](https://user-images.githubusercontent.com/5117473/43305358-19bb1152-913d-11e8-8324-c5819c2c3a3f.png)

Also, I'm not sure what version you'd want on this change. Probably 1.1.0 due to the UI changes, but maybe even 2.0 since the changes are quite different, albeit now very closely match the native Android UI.

Cheers! 😁

jstheoriginal commented:
Share Link:

https://expo.io/--/to-exp/exp%3A%2F%2Fwr-gu2.jstheoriginal.example.exp.direct%3A80

jstheoriginal commented:
Any chance of reviewing this?

peterpme commented:
Hey @jstheoriginal , sorry was out of town. That Snack link dosen't work. Mind putting up another one?

Thanks!

peterpme commented on ActionSheet.ios.js:
```diff
@@ -5,10 +5,20 @@ import { ActionSheetIOS, View } from 'react-native';
 export default class ActionSheet extends React.Component {
   render() {
-    return <View style={{ flex: 1 }}>{React.Children.only(this.props.children)}</View>;
+    return (
```
any reason you changed the formatting here? Seems like the closing tag has an extra space. Do you mind running this through `prettier`?

jstheoriginal replied:
I reverted the change. I couldn't find any prettier settings anywhere so I'm not sure what prettier options to run it through. If you know what they are, I'll gladly run it.

peterpme commented on ActionSheet.ios.js:
```diff
@@ -5,10 +5,20 @@ import { ActionSheetIOS, View } from 'react-native';
-    ActionSheetIOS.showActionSheetWithOptions(options, onSelect);
+    const iosOptions = {
+      ...options,
+      title: options.title === null ? undefined : options.title,
```
do we need an explicit null check here? What happens if you just set it to options.title? Does it fail?

jstheoriginal replied:
Yeah I had a native crash on iOS. I’ll see what happens again and post the crash.

peterpme replied:
ok cool! In that case you can also do `options.title == null` which will cover both undefined / null cases. double equals is ok here and there 😉

jstheoriginal replied:
Ok I wasn’t sure about the == being ok. 

So in that case, are you good with 

`title: options.title || undefined,`

?

jstheoriginal replied:
It was strange. When title was undefined, it didn’t crash, but did crash only if title was explicitly set to null (which is why I coded it like I did so it was clear it was just when exactly null).

peterpme commented on example/App.js:
```diff
@@ -16,45 +21,113 @@ export default class AppContainer extends React.Component {
+    selectedIndex: null,
+  };
+
+  _updateSelectionText = (selectedIndex: number) => {
```
i'm not sure flow is supported here so you can remove this annotation

peterpme commented on example/App.js:
```diff
@@ -16,45 +21,113 @@ export default class AppContainer extends React.Component {
+    this.setState({ selectedIndex });
+  };
+
+  _renderButton = (title: string, onPress: () => void) => (
```
you can remove annotation

peterpme commented on example/App.js:
```diff
@@ -16,45 +21,113 @@ export default class AppContainer extends React.Component {
-          <Text style={{ textAlign: 'center' }}>
-            Hello! This is a simple example app to demonstrate @expo/react-native-action-sheet.
+      <View style={{ flex: 1 }}>
+        <ScrollView style={styles.scrollView} contentContainerStyle={styles.contentContainer}>
```
any reason we need a scroll view to render buttons?

jstheoriginal replied:
Yeah it’s in case there are too many options to fit, as it has a max height. This was put in by the previous person who made the original PR.

I can remove it, I just need some direction on how we would handle options that won’t fit on shorter devices.

peterpme commented on example/App.js:
```diff
@@ -16,45 +21,113 @@ export default class AppContainer extends React.Component {
           </Text>
-        </View>
-
-        <Entypo.Button name="code" backgroundColor="#3e3e3e" onPress={this._onOpenActionSheet}>
```
i think its ok to leave the button as is. You can actually apply the `marginBottom` style directly to `Entypo.Button` as well

jstheoriginal replied:
I remember now why I had to wrap this in a view (after doing what you suggested). For some reason, the Entypo button applies the margin inside the button rather than outside. :/

jstheoriginal replied:
For proof, here's what happens when i applied the margin on the Entypo.Button:

![img_0060](https://user-images.githubusercontent.com/5117473/44425775-72fec280-a552-11e8-9797-dc65d39e1ded.PNG)

peterpme commented on example/App.js:
```diff
@@ -1,9 +1,14 @@
+const saveIcon = require('./img/baseline_save_black_24dp.png');
+const shareIcon = require('./img/baseline_share_black_24dp.png');
+const cancelIcon = require('./img/baseline_cancel_black_24dp.png');
+const deleteIcon = require('./img/baseline_delete_black_24dp.png');
```
can we possibly utilize the icons that are shipped with Expo? We're already using Entypo, we could utilize those or use SimpleLineIcons for iOS and Material for Android

jstheoriginal replied:
done like dinner.

peterpme commented on example/App.js:
```diff
@@ -16,45 +21,113 @@ export default class AppContainer extends React.Component {
+          {this._renderButton('Icons', this._onOpenActionSheet(false, false, true, false, false))}
+          {this._renderButton('Title, Message, & Icons', this._onOpenActionSheet(true, true, true, false, false))}
+          {this._renderButton('Use Separators', this._onOpenActionSheet(true, false, true, true, false))}
+          {this._renderButton('Custom Styles', this._onOpenActionSheet(true, true, true, false, true))}
```
do you mind inlining these as components so its a little easier to understand?

```js
<ActionSheetButton title="Options Only" />
<ActionSheetButton title="Custom Styles" withTitle withMessage withIcons />
```

peterpme commented on example/App.js:
```diff
@@ -16,45 +21,113 @@ export default class AppContainer extends React.Component {
-    let destructiveButtonIndex = 0;
-    let cancelButtonIndex = 2;
+    const options = ['Delete', 'Save', 'Share', 'Cancel'];
+    const icons = withIcons ? [deleteIcon, saveIcon, shareIcon, cancelIcon] : null;
```
re: icons - it would be great to utilize icons already shipped in the project

jstheoriginal replied:
For sure! I haven’t used expo so I didn’t realize it had icons already. 🎉

jstheoriginal commented:
I believe everything is addressed that could be!

I added another new prop for Android: `tintIcons`. By default, this library is applying the same tint color to the icons as the option text color. This didn't allow the use of full color image icons, so now if you set tintIcons to false, the icons will render at the color of the source image.

While incorporating the vector-icons, i decided to make it possible to pass in pre-rendered nodes into the icons array, so you can pass the appropriate pre-rendered icon without needing to use the async `getImageSource` function on vector-icons--you can now just pass in an array of the icon components instead of image sources, if you want. 🎉 

Updated Share Link: https://expo.io/--/to-exp/exp%3A%2F%2Fwr-gu2.jstheoriginal.example.exp.direct%3A80

Here's an example showing the use of a full color image for an icon.

![screenshot_1534879334](https://user-images.githubusercontent.com/5117473/44425558-d0deda80-a551-11e8-88cc-3ba699355e44.png)

JulianKingman commented:
@peterpme is there anything currently blocking the merging of this? @jstheoriginal maybe the icon stuff should be a separate PR? Every now and then I check back on this PR, I'm looking forward to its merging.

jstheoriginal commented:
> @peterpme is there anything currently blocking the merging of this? @jstheoriginal maybe the icon stuff should be a separate PR? Every now and then I check back on this PR, I'm looking forward to its merging.

everything including the icons has been addressed. 😃

peterpme commented on ActionSheet.android.js:
```diff
@@ -87,19 +96,24 @@ class ActionGroup extends React.Component {
-
+      const defaultColor = tintColor
+        ? tintColor
+        : (textStyle || {}).color || BLACK_87PC_TRANSPARENT;
```
can you set the defaultProp of a textStyle to an empty object so we don't have to fallback to that here?

jstheoriginal replied:
OK I remember now why I couldn't do it the way you suggest. The parameters for showing an ActionSheet are as follows:

```
showActionSheetWithOptions(
    options: ActionSheetOptions,
    onSelect: (i: number) => void,
    onAnimateOut: () => void
  ) {
```

So I did try to add it to defaultProps, but it just crashes saying textStyle is null. In the render function for ActionSheet, it automatically always uses the passed in options object, which I can't default since that entire object is user-supplied.

jstheoriginal replied:
And I just noticed that my defaultProp for `tintIcons` that I set to true doesn't work for the same reason. So it'll need to default to false/undefined.

jstheoriginal replied:
derp. I'm dumb. I can still do it, just not using defaultProps. 😅 

`textStyle={options.textStyle || {}}`

peterpme commented on ActionSheet.android.js:
```diff
@@ -87,19 +96,24 @@ class ActionGroup extends React.Component {
+      const defaultColor = tintColor
+        ? tintColor
+        : (textStyle || {}).color || BLACK_87PC_TRANSPARENT;
+      const color = i === destructiveButtonIndex ? '#d32f2f' : defaultColor;
```
you've placed some constants up top for colors, can you do the same for `#d32f2f`?

peterpme commented on ActionSheet.android.js:
```diff
@@ -87,19 +96,24 @@ class ActionGroup extends React.Component {
+            resizeMode="contain"
+            style={iconStyle}/>;
+        } else {
+          iconElement = <View style={styles.icon}>{iconSource}</View>;
```
what exactly is `iconSource` in this case? an React element?

jstheoriginal replied:
Either a required image source, or a React element, correct.

peterpme commented on ActionSheet.android.js:
```diff
@@ -87,19 +96,24 @@ class ActionGroup extends React.Component {
+          const iconStyle = [styles.icon, { tintColor: tintIcons ? color : null }];
+          iconElement = <Image
+            fadeDuration={0}
+            source={iconSource}
```
in this case its a `require`'d image?

jstheoriginal replied:
Correct.

peterpme commented on ActionSheet.android.js:
```diff
@@ -87,19 +96,24 @@ class ActionGroup extends React.Component {
-          iconStyle.push({ tintColor: textStyle.color })
+      if (iconSource) {
+        if (typeof iconSource === 'number') {
+          const iconStyle = [styles.icon, { tintColor: tintIcons ? color : null }];
```
what does tintIcons have to do with setting the color here

This is how it reads:
If `tintIcons` exist, set it to color. Otherwise set it to null?

jstheoriginal replied:
Well `tintIcons` is a Boolean prop you can set if you want to apply whatever colour the text will be to the icons as well. If someone doesn’t want to change the tint (ie they’ve exported the icons at a specific colour they want already, they can set this to false.

Maybe a different name for this prop instead?

jstheoriginal replied:
need to know if the name needs to be better, or is fine as-is. I did add all of the new ones to the documentation.

zibs replied:
Seems clear to me with the added documentation.

peterpme commented on ActionSheet.android.js:
```diff
@@ -110,26 +124,74 @@ class ActionGroup extends React.Component {
+    return (
+      <View>
+        <View style={[styles.titleContainer, { paddingBottom: showSeparators ? 24 : 16 }]}>
+          {title ? <Text style={[styles.title, titleTextStyle]}>{title}</Text> : null}
```
you can do `title && <Text /> ...` here instead of a ternary

JulianKingman replied:
@peterpme `title && <Text />` doesn't work in the infrequent case that `title === 0`, in which case it produces an error that it needs to be contained in a `<Text>` element, for which reason I always use ternary in react.

jstheoriginal replied:
@JulianKingman 😁 yup that’s why we always do ternary as well. We’ve been burnt by not explicitly returning null in the past where I work.

maxaggedon replied:
I usually write `{!!title && <Text />}` which is shorter

jstheoriginal replied:
Need to know which way is preferred...

zibs replied:
@jstheoriginal use @maxaggedon method, that should be fine or cast it to a boolean `Boolean(title) && <Text />`

peterpme commented on ActionSheet.android.js:
```diff
@@ -110,26 +124,74 @@ class ActionGroup extends React.Component {
+          {title ? <Text style={[styles.title, titleTextStyle]}>{title}</Text> : null}
+          {message ? <Text style={[styles.message, messageTextStyle]}>{message}</Text> : null}
+        </View>
+        {showSeparators ? this._renderRowSeparator('title') : null}
```
you can do `showSeparators && this._renderRow ...` here

JulianKingman replied:
as long as showSeparators will never === 0, this works.

jstheoriginal replied:
leave this or update to something like `!!showSeparators && this._renderRow ...`?

zibs replied:
so long as showSeparators is always a boolean (or else it'll crash android) you can do ` && this._renderRow` - even `Boolean(showSeparators) && this._renderRow` would work.

jstheoriginal replied:
showSeparators is an optional boolean prop, so it wouldn't actually work as suggested. So i'll do the !! in front to cast the boolean or undefined to boolean.

peterpme commented:
Hi everyone,

Thank you for reminding me. This is a really big PR and a lot of people rely on react-native-action-sheet so I just have to be conscious of any choices we make. 

If there are API adjustments, we'll have to take some time thinking about semver.

Niryo commented:
This is an incredible PR, I hope it's going to be merged soon. 
It is one of the most basic things I would expect from an action sheet. My use-case for example is to open an action sheet on long pressing an item in a list. without the title, there is no way to indicate the user that he indeed pressed on the correct item.

Thank you @jstheoriginal :) If I can help with something let me know.

jstheoriginal commented:
I’ll get to these final fix ups soon...it’s thanksgiving weekend up in the Canada this weekend, eh. 😅🇨🇦 I’ve got Monday off from work if I can’t get to it before then. 🦃

Niryo commented:
@peterpme ping (please don't let this PR gone away)

peterpme commented:
an you do me a favor @Niryo and get an example of this working on https://snack.expo.io? I won't forget about it but will need a hand testing 😄

Niryo commented:
@peterpme sure, I will try to create an example soon.

Niryo commented:
Ok, I don't have any experience with this Snack thing, and it didn't play well with my attempts to use jstheoriginal repo as a dependency in the package JSON, so I ended up uploading directly the files from jstheoriginal's clone.
It also did not play well with the android/ios file ending, so I changed one line in ActionSheetProvider (line 5 in the snack). 
this is the snack: https://snack.expo.io/SkMqEYLs7

zibs commented:
@peterpme is there anything to speed this up? I can jump in if possible perhaps.

jstheoriginal commented:
All changes pushed. I also added a new `separatorStyle` Android-only prop, in case that needs to be tweaked.

jstheoriginal commented:
I believe this warrants a 2.0 release since it's not close to the old look. I'll prepare a commit with that change as well.

Does this project referenced in the readme need to be updated as well? or does it change based on source?

>Try it in Expo: https://expo.io/@community/react-native-action-sheet-example

jstheoriginal commented:
Should I also be adding some of these new props to the provider, like in https://github.com/expo/react-native-action-sheet/commit/dfe724247ceec13fa34a712dc5a661157b3b2fe9#diff-f2a284b50111aa631ee498cfc2b01ff4?

The benefit of this is that you can set stylistic properties and such once, correct?

If so, I think I should add these there as well:

- tintIcons
- tintColor
- titleTextStyle
- messageTextStyle
- showSeparators
- separatorStyle

Agreed?

jstheoriginal commented:
Alright. I've added those style-type props to the Provider, upped the version to 2.0.0, and did some cleanup of the App.js.

Here's an updated Snack: https://snack.expo.io/@jstheoriginal/reactnativeactionsheet

zibs commented:
I'm running the updated snack on Android + iOS and it's working. I can't seem to adjust the tintColor of the icons on the snack, but I assume that's because we're just rendering those vector icon components, and not an image component?

jstheoriginal commented:
Correct. `tintColor` would only work for images provided.

I noted this in the documentation for `tintIcons`, so I think it's clear as-is: 
>If you provide your own nodes rather than an image source, you will need to tint them appropriately.

agreed?

zibs commented:
Yep that sounds good to me - so you can essentially just DIY like you are with `const icon = name => <MaterialIcons key={name} name={name} size={24} />` and add your desired style to it so it'd  `const icon = name => <MaterialIcons key={name} name={name} size={24} style={{tintColor: 'pink'}} />`

jstheoriginal commented:
Meh...it's too ambiguous. I'll change it to this:

>If you provide your own nodes/pre-rendered icons rather than required images in the `icons` array, you will need to tint them appropriately before providing them in the array of icons; `tintColor` will not be applied to icons unless they are images from a required source.

jstheoriginal commented:
Yup.

zibs commented:
Woot! hey @peterpme - I think it's ready for you take a look and give it a go.

jstheoriginal commented:
Also, here are some release notes. I think it covers what's changed well enough, but feel free to tweak them:

```
This release is all about making the ActionSheet on Android look and feel as close to the native material ActionSheet as possible!

### ActionSheet on Android Changes

- the ability to provide a title and message above the options just like on iOS (_props:_ `title` and `message`)
- the ability to set custom text styles for title and message (_props:_ `titleTextStyle` and `messageTextStyle`)
- the ability to provide pre-rendered components for icons. i.e.: you could provide an array of vector-icons as done in the example project.
- the ability to use your own full-colour icons (set `tintIcons` to `false` and the text tint won't be applied to your icons, allowing them to render as provided)
- the ability to set a custom tint color for option text and icons (if `tintIcons` is `true`) (_prop:_ `tintColor`)
- the ability to show separators between options (default is `false`), and set the separator style to modify it from the default style. (_prop:_ `showSeparators` and `separatorStyle`)

Visit the [Android-Only Props](https://github.com/expo/react-native-action-sheet#android-only-props) section of the README for descriptions and defaults of the props.

Note: The Android additions above do not apply to iOS as iOS's implementation matches the native look of ActionSheet on iOS. (see [ActionSheetIOS documentation](https://facebook.github.io/react-native/docs/actionsheetios.html) for what is available on iOS).

### Example App Changes

- it uses SafeAreaView to layout properly on phones with notches
- it provides examples for multiple option configurations to see how they look
```

This release is all about making the ActionSheet on Android look and feel as close to the native material ActionSheet as possible!

### ActionSheet on Android Changes

- the ability to provide a title and message above the options just like on iOS (_props:_ `title` and `message`)
- the ability to set custom text styles for title and message (_props:_ `titleTextStyle` and `messageTextStyle`)
- the ability to provide pre-rendered components for icons. i.e.: you could provide an array of vector-icons as done in the example project.
- the ability to use your own full-colour icons (set `tintIcons` to `false` and the text tint won't be applied to your icons, allowing them to render as provided)
- the ability to set a custom tint color for option text and icons (if `tintIcons` is `true`) (_prop:_ `tintColor`)
- the ability to show separators between options (default is `false`), and set the separator style to modify it from the default style. (_prop:_ `showSeparators` and `separatorStyle`)

Visit the [Android-Only Props](https://github.com/expo/react-native-action-sheet#android-only-props) section of the README for descriptions and defaults of the props.

Note: The Android additions above do not apply to iOS as iOS's implementation matches the native look of ActionSheet on iOS. (see [ActionSheetIOS documentation](https://facebook.github.io/react-native/docs/actionsheetios.html) for what is available on iOS).

### Example App Changes

- it uses SafeAreaView to layout properly on phones with notches
- it provides examples for multiple option configurations to see how they look

jstheoriginal commented:
This is what the example app now looks like:

![ivborw0kggoaaaansuheugaabs0aaao4cayaaac8jok aaabgmldq1bzukdcielfqzyxoty2ltiu-2](https://user-images.githubusercontent.com/5117473/48317619-358e5c00-e5ba-11e8-916c-c0a6688af6d0.PNG)

Devyup31 commented:
Solid work. Is there something holding this back from merging? Happy to assist. @peterpme

peterpme commented:
I’m out of town until the 27th.

A working example on snack.expo.io is what we need to make sure it’s all working right on iOS and Android.

Thanks!

Devyup31 commented:
This seems to be the latest Snack. (@jstheoriginal correct?)
https://snack.expo.io/@jstheoriginal/reactnativeactionsheet

Ran it and seemed fine by me 👌🏼

peterpme commented:
Awesome! iOS? Android? Can we get folks on both sides to confirm? Thanks!

jstheoriginal commented:
That’s the correct snack. 

And iOS runs fine for me on my device right now. I tested it with Android, but if someone with Android could run that snack and confirm all good, we should be good to go.

Niryo commented:
@jstheoriginal looks good on xiaomi A1, pure android 8.1:)

jstheoriginal commented:
Nice. Thanks for confirming! 

Maybe we can get this in before the 2 year anniversary of the original pull request's submission (Dec 2nd, 2016 😄).

jstheoriginal commented:
I did have a question about the Try it out link https://expo.io/@community/react-native-action-sheet-example in the readme...does the source of that link update itself automatically once this gets a new release?

Devyup31 commented:
Yup. Tested on iOS 12 and Android 8.0 (Galaxy S8)

jermaine commented:
> Awesome! iOS? Android? Can we get folks on both sides to confirm? Thanks!

@peterpme @jstheoriginal All good on Android 5.1, Android 9.0, iOS 12.1 👌🏽
Good to go by me  🚀🚀

peterpme commented:
Ok everyone, sounds like we have ourselves a winner! Thanks for everyone involved:

- @jstheoriginal
- @jermaine 
- @Devyup31 
- @Niryo  
- @zibs 
- @maxaggedon 

I'll merge, test and cut a new release tonight 😍

jstheoriginal commented:
Woohoo the longest open PR ever ends! 😄 Thanks @peterme!

peterpme commented:
Hey everyone,

bad news. This error pops when running the example using version `2.0.1-0` (prerelease)

```"JSON value '<null>' of type NSNull cannot be converted to a UIColor. Did you forget to call processColor() on the JS side?```

You can try the example [here](https://github.com/expo/react-native-action-sheet/pull/94/files)

peterpme commented:
i actually changed it frmo orange to #000 for that very reason hoping it would fix it, haha.

@jstheoriginal can you update your snack to use `@expo/react-native-action-sheet` directly possibly?

jstheoriginal commented:
How can i run your example @peterpme?

jstheoriginal commented:
Ok i updated my snack to use 2.0.1-0 and it's not crashing. 🤔 

https://snack.expo.io/@jstheoriginal/reactnativeactionsheet

JulianKingman commented:
I believe this line is the problem: https://github.com/expo/react-native-action-sheet/blob/master/ActionSheetProvider.js#L24

`tintColor`, not `titleTextStyle`

jstheoriginal commented:
That PR ^ should fix this.

peterpme commented:
Just published 2.0.0-1 to test this! Thanks everyone!

peterpme commented:
Hey everyone, 2.0.1 is officially out!! Thank you for the help!!!
